pkg/plugins/leader/plugin.go

```diff
@@ -0,0 +1,41 @@
+package leader
+
+import (
+	"time"
+
+	"cirello.io/pglock"
+	"github.com/pkg/errors"
+
+	"github.com/Kong/kuma/pkg/config/core/resources/store"
+	core_runtime "github.com/Kong/kuma/pkg/core/runtime"
+	"github.com/Kong/kuma/pkg/core/runtime/component"
+	common_postgres "github.com/Kong/kuma/pkg/plugins/common/postgres"
+	leader_memory "github.com/Kong/kuma/pkg/plugins/leader/memory"
+	leader_postgres "github.com/Kong/kuma/pkg/plugins/leader/postgres"
+)
+
+func NewLeaderElector(b *core_runtime.Builder) (component.LeaderElector, error) {
+	switch b.Config().Store.Type {
+	case store.PostgresStore:
+		db, err := common_postgres.ConnectToDb(*b.Config().Store.Postgres)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not connect to postgres")
+		}
+		client, err := pglock.New(db,
+			pglock.WithLeaseDuration(5*time.Second),
```

Comment:
Does this setting mean that lock will be acquired only on 5 sec and released after it? I'm not sure that correctly understand that postgres mechanism

Reply:
No. Heartbeat extends the lease, when the Heartbeat stops then after 5s other instance will take a lead